app/controllers/users_email_controller.rb

```diff
@@ -3,24 +3,16 @@
 class UsersEmailController < ApplicationController
   requires_login only: %i[index update]
 
-  skip_before_action :check_xhr,
-                     only: %i[
-                       confirm_old_email
-                       show_confirm_old_email
-                       confirm_new_email
-                       show_confirm_new_email
-                     ]
+  skip_before_action :check_xhr, only: %i[show_confirm_old_email show_confirm_new_email]
 
   skip_before_action :redirect_to_login_if_required,
                      only: %i[
-                       confirm_old_email
                        show_confirm_old_email
-                       confirm_new_email
                        show_confirm_new_email
+                       confirm_old_email
+                       confirm_new_email
                      ]
 
-  before_action :require_login, only: %i[confirm_old_email show_confirm_old_email]
```

Comment:
Should these routes be available to anonymous users? I pulled this locally and can reach the URL, but after clicking the "Confirm" button, the app returns a 403: 

<img width="1424" alt="image" src="https://github.com/discourse/discourse/assets/368961/2bdf7dcf-89d1-4d4d-9b72-4776f899bd52">

We might as well block access a step earlier.

Reply:
confirm-new-email has always been accessible to anon, so I figured it make sense to do the same with confirm-old-email. 

This error was a CSRF thing because I passed the wrong opts to `ajax()`. Should be good now - thanks 👌